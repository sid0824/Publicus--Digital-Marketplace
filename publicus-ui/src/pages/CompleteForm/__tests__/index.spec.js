import toJson from 'enzyme-to-json';
import React from "react";
import {shallow, configure} from 'enzyme';
import { message, Form } from 'antd';
import Adapter from 'enzyme-adapter-react-16';
import { CompleteForm } from "../index";
import {API} from "aws-amplify";
import Forms from '../../../fixtures/Forms';

configure({ adapter: new Adapter() });


const mockValues = {
  1: 'VALUE_1',
  2: 'VALUE_2'
};

const mockForm = {
  validateFields: jest.fn().mockImplementation(async (func) => func(null, mockValues))
};

const mockLocation = {
  search: "?surveyId=4fe4b2c0-f4a5-11e9-87b0-6785951dc893&creatorId=ap-southeast-2:91ecfa27-a7c0-45a8-9f45-c93885409ca7"
};

const mockPush = jest.fn();
const mockHistory = {
  push: mockPush
};

describe('Complete Form', () => {

  beforeEach(() => {
    API.get = jest.fn().mockResolvedValue(Forms[2]);
    API.post = jest.fn().mockResolvedValue(null);
    message.success = jest.fn()
    message.error = jest.fn()
  });

  afterEach(() => {
    API.get.mockReset();
    API.post.mockReset();
    message.success.mockReset();
    message.error.mockReset();
    mockPush.mockReset();
  });

  test('should render complete form page', (done) => {
    const wrapper = shallow(
      <CompleteForm
        form={mockForm}
        location={mockLocation}
        history={mockHistory}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    setTimeout(() => {
      expect(toJson(wrapper)).toMatchSnapshot();
      done();
    })
  });

  test('should give error and redirect on invalid form', async (done) => {
    API.get.mockImplementation(() => {
      throw new Error()
    });
    shallow(
      <CompleteForm
        form={mockForm}
        location={{
          search: "?invalid=url"
        }}
        history={mockHistory}
      />
    );
    setTimeout(() => {
      expect(message.error).toHaveBeenCalledTimes(2);
      expect(message.error.mock.calls[0]).toMatchSnapshot();
      expect(mockPush.mock.calls[0][0]).toMatchSnapshot();
      done();
    })
  });

  test('should submit results of complete form and redirect', async (done) => {
    const wrapper = shallow(
      <CompleteForm
        form={mockForm}
        location={mockLocation}
        history={mockHistory}
      />
    );
    setTimeout(async () => {
      await wrapper.instance().onSubmit({ preventDefault: () => {}});
      setTimeout(() => {
        expect(message.success.mock.calls[0]).toMatchSnapshot();
        expect(mockPush.mock.calls[0][0]).toMatchSnapshot();
        expect(API.post.mock.calls[0]).toMatchSnapshot();
        done();
      }, 1100)
    })
  });


});