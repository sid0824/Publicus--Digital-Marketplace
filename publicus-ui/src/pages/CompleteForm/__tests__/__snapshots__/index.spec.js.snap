// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Complete Form should give error and redirect on invalid form 1`] = `
Array [
  "Invalid Form. Redirecting",
]
`;

exports[`Complete Form should give error and redirect on invalid form 2`] = `"/"`;

exports[`Complete Form should render complete form page 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "height": "100%",
      "justifyContent": "center",
      "paddingTop": 200,
      "width": "100%",
    }
  }
>
  <Icon
    style={
      Object {
        "fontSize": 70,
        "marginTop": 8,
      }
    }
    type="loading"
  />
</div>
`;

exports[`Complete Form should render complete form page 2`] = `
<div
  style={
    Object {
      "marginLeft": "calc(50% - 250px)",
      "marginRight": "calc(50% - 250px)",
      "marginTop": 10,
    }
  }
>
  <h1>
    After-Edit
  </h1>
  null
  <br />
  <br />
  <Form
    colon={true}
    hideRequiredMark={false}
    layout="horizontal"
    onSubmit={[Function]}
  >
    What is Your... ?
    <FormComponent
      form={
        Object {
          "validateFields": [MockFunction],
        }
      }
      id="1"
      question={
        Object {
          "prompt": "What is Your... ?",
          "rules": Array [
            "required",
          ],
          "type": "textInput",
        }
      }
    />
    What is Your... ?
    <FormComponent
      form={
        Object {
          "validateFields": [MockFunction],
        }
      }
      id="2"
      question={
        Object {
          "defaultValue": false,
          "prompt": "What is Your... ?",
          "rules": Array [
            "required",
          ],
          "type": "switch",
        }
      }
    />
    <FormItem
      hasFeedback={false}
    >
      <Button
        block={false}
        ghost={false}
        htmlType="submit"
        loading={false}
        style={
          Object {
            "marginRight": 5,
          }
        }
        type="primary"
      >
        Submit
      </Button>
    </FormItem>
  </Form>
</div>
`;

exports[`Complete Form should submit results of complete form and redirect 1`] = `
Array [
  "Saved Successfully",
]
`;

exports[`Complete Form should submit results of complete form and redirect 2`] = `"/"`;

exports[`Complete Form should submit results of complete form and redirect 3`] = `
Array [
  "prod-publicus-app-api",
  "/results/ba665a40-f58b-11e9-9b29-99ffbf733bb5",
  Object {
    "body": Object {
      "creatorId": "ap-southeast-2:91ecfa27-a7c0-45a8-9f45-c93885409ca7",
      "results": Object {
        "1": "VALUE_1",
        "2": "VALUE_2",
      },
    },
  },
]
`;
